R/get_data.R

```diff
@@ -156,14 +161,24 @@ bcdc_get_data.bcdc_record <- function(record, resource = NULL, ...) {
     # cat(glue::glue("bcdc_get_data('{x}', resource = '{id_choice}')"),"\n")
     query <- bcdc_query_geodata(record = record_id, ...)
     return(collect(query))
+  }
+
+  if(dup_choice){
+    ext <- gsub(".*\\((.*)\\).*", "\\1", name_choice)
+    name <- gsub("\\s*\\([^\\)]+\\)","", name_choice)
+
+    resource <- resource_df[resource_df$name == name & resource_df$format == ext, , drop = FALSE]
+    id_choice <- resource_df$id[resource_df$name == name & resource_df$format == ext]
+
   } else {
     resource <- resource_df[resource_df$name == name_choice, , drop = FALSE]
     id_choice <- resource_df$id[resource_df$name == name_choice]
```

Comment:
Is it possible to simplify this whole thing to:

```r
resource <- resource_df[choice_input, , drop = FALSE]
id_choice <- resource_df$id[choice_input]
```

i.e., the row number of the `resource_df` should always match up with the choices presented, right? We've already subsetted the `resource_df` with `bcdata_available` so there should be the same number of rows in `resource_df`, and their indices should match up with the menu choices... right?

Reply:
Thinking about this more, and trying it out on a few records, I think it's the way to go. Simplifies it a lot, and gets rid of the conditional and the grepping.

Reply:
Great idea.